src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+let rightButtonClicked = false;
+let leftButtonClicked = false;
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    leftButtonClicked = true;
+    resolve('First promise was resolved');
+  });
+
+  if (!leftButtonClicked) {
+    setTimeout(() => {
+      // eslint-disable-next-line
+      reject(`First promise was rejected`);
+    }, 3000);
+  }
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      leftButtonClicked = true;
+    }
+
+    if (e.button === 2) {
+      rightButtonClicked = true;
+    }
+
+    if (rightButtonClicked && leftButtonClicked) {
+      resolve('Third promise was resolved');
+    }
+  });
+});
```

Comment:
The third promise logic is correct in checking for both left and right button clicks. However, ensure that the state of `leftButtonClicked` and `rightButtonClicked` is reset if needed, as they persist across multiple events.